src/newcomponents/info/ChatInfo.js

```diff
@@ -0,0 +1,141 @@
+import { getConverastionById } from "@store/values/Conversations";
+import { getCurrentUser } from "@store/values/CurrentUser";
+import { selectParticipantsEntities } from "@store/values/Participants";
+import { useLocation } from "react-router-dom";
+import { useMemo } from "react";
+import { useSelector } from "react-redux";
+
+import "@newstyles/info/ChatInfo.css";
+
+import { ReactComponent as AddParticipants } from "@newicons/AddParticipants.svg";
+import { ReactComponent as Close } from "@newicons/actions/CloseGray.svg";
+import { ReactComponent as ImageBig } from "@newicons/media/ImageBig.svg";
+
+export default function ChatInfo() {
+  const { pathname, hash } = useLocation();
+
+  const participants = useSelector(selectParticipantsEntities);
+  const currentUser = useSelector(getCurrentUser);
+  const selectedConversation = useSelector(getConverastionById);
+
+  const isCurrentUserOwner = useMemo(() => {
+    if (!currentUser || !selectedConversation) {
+      return false;
+    }
+    return currentUser._id === selectedConversation.owner_id?.toString();
+  }, [currentUser, selectedConversation]);
+
+  window.onkeydown = function (event) {
+    event.keyCode === 27 && removeAndNavigateSubLink(pathname + hash, "/info");
+    event.keyCode === 13 && event.preventDefault();
+  };
+
+  //HOLD FOR RIGHT CLICK CONTEXT MENU
+  // const deleteChat = async () => {
+  //   const isConfirm = window.confirm(`Do you want to delete this chat?`);
+  //   if (isConfirm) {
+  //     try {
+  //       await api.conversationDelete({ cid: selectedCID });
+  //       dispatch(clearSelectedConversation());
+  //       dispatch(removeChat(selectedCID));
+  //       navigateTo("/");
+  //     } catch (error) {
+  //       showCustomAlert(error.message, "warning");
+  //     }
+  //   }
+  // };
+
+  const participantsList = useMemo(() => {
+    if (!selectedConversation?.participants || !currentUser) {
+      return null;
+    }
+
+    return selectedConversation.participants.map((uId) => {
+      const userObject = participants[uId];
+      if (!userObject) {
+        return null;
+      }
+
+      const isOwner =
+        userObject._id === selectedConversation.owner_id?.toString();
+
+      //HOLD FOR RIGHT CLICK CONTEXT MENU
+      // const deleteUser = async (event) => {
+      //   event.stopPropagation();
+
+      //   if (!window.confirm(`Do you want to delete this user?`)) {
+      //     return;
+      //   }
+
+      //   const requestData = {
+      //     cid: selectedCID,
+      //     participants: { remove: [u._id] },
+      //   };
+      //   await api.conversationUpdate(requestData);
+      //remove user form participants field - redux
+      // };
+
+      return (
+        <ParticipantInChat
+          key={uId}
+          userObject={userObject}
+          isOwner={isOwner}
+        />
+      );
+    });
+  }, [selectedConversation, participants, currentUser]);
+
+  const participantsCount = useMemo(
```

Comment:
It can be simply calculated as `const participantsCount = participantsList.length`, w/o additional `useMemo`